keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,610 @@
+any component that run on Kubernetes.
+This will also help external consumers of the e2e test framework by reducing
+the amount of dependency management they have to do now when importing the e2e
+test framework.
+
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The e2e test framework is a Kubernetes component that has been evolving
+organically to meet all the needs that contributors have in order to test
+enhancements and behaviours of Kubernetes.
+The SIG Testing sub-project
+[Testing Commons](https://github.com/kubernetes/community/tree/master/sig-testing#testing-commons)
+has been looking at ways of improving the user experience for contributors to
+the Kubernetes community.
+This KEP is part of that goal.
+
+We foresee the following benefits coming out of this work:
+
+1. Maintainability: the e2e test framework lacks a general API - there has been
+  little discussion on what parts of the framework do what.
+  From this work we plan to organize the code that is present and use only
+  external libraries and APIs available for e2e tests.
+  This decoupling will ensure the e2e test framework packages are better
+  contained allowing for e2e tests to have a more standard flow.
+2. External consumption: The Kubernetes ecosystem has grown considerably since
+  the creation of the e2e test framework. Some projects, such as the cloud
+  providers and CSI extensions, began within the core Kubernetes codebase and
+  have been migrated out or are in the process of being migrated out. We want
+  to provide them with the same set level of functionality they had in core
+  without having to deal with the problems of importing the entirety of core
+  Kubernetes.
+
+There are two main components to the e2e test framework: its configuration
+(used for setting up and running e2e tests) and its set of packages and utility
+functions.
+For this work, we plan on focusing on the latter.
+The entrypoint into the framework, the configuration part, will stay as is.
+Only the utility functions will be refactored.
+What this means for consumers, is that the functions offered via the e2e test
+framework will change but not the way e2e tests are setup and executed.
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+* Decouple the e2e test framework and all of its subpackages from any
+  Kubernetes internal packages (i.e., pkg, other e2e packages).
+* Impose import restrictions to prevent the use of core Kubernetes packages.
+* Allow the e2e test framework to be easily used and without requiring the
```

Comment:
this is the only goal that seems related to the title of the kep. the rest of these seem like admirable but orthogonal goals.

Reply:
yes. I was thinking that explicitly calling the move to stage would be the biggest change.
All other work is just "maintainability" and needed even if the e2e test framework was not to land in staging.